### PR TITLE
Update Cairo to version 1.18.0

### PR DESCRIPTION
**Pull request by danyeaw**
I had to patch Cairo to get this to compile, and I submitted the patch upstream:

https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/520

### CODE REVIEW

**Review by pbor — APPROVED** [review]

**Comment by danyeaw** [comment]
I'm not sure why the build of Pango is failing :thinking:

**Comment by pbor** [comment]
```
testserialize.c.obj : error LNK2019: unresolved external symbol pango_cairo_fc_font_map_get_type referenced in function install_fonts
```

**Comment by danyeaw** [comment]
Hey @pbor, yup, I'm not sure if this is an issue with Cairo, Pango, or Fontconfig yet. Maybe an incompatibility with Pango after the Cairo update?

**Comment by pbor** [comment]
Yeah, not sure, I just pasted the line so we do not need to fish it in the log each time, but I am not sure how to fix :)

**Comment by pbor** [comment]
In the mean time, we should start pushing https://github.com/wingtk/gvsbuild/pull/1086/commits/1507412fa9b7b1a191aab41ceeeeb8cafb566ae2

**Comment by danyeaw** [comment]
@pbor It looks like upgrading Pango to 1.51 did the trick

File: gvsbuild/patches/cairo/0001-fix-alloca-unresolved.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+===================================================================
+diff --git a/src/cairo-colr-glyph-render.c b/src/cairo-colr-glyph-render.c
+--- a/src/cairo-colr-glyph-render.c	(revision 459d19ef1520d39078038ecaa840efca0194e053)
++++ b/src/cairo-colr-glyph-render.c	(date 1696094155649)
+@@ -43,6 +43,10 @@
+ #include <stdio.h>
+ #include <string.h>
+ 
++#ifdef _MSC_VER
++#include <malloc.h>
```
Comment (nacho) [inline gvsbuild/patches/cairo/0001-fix-alloca-unresolved.patch:16]: do we have a ticket upstream to track this?

Comment (danyeaw) [reply gvsbuild/patches/cairo/0001-fix-alloca-unresolved.patch:16]: Yup, it is already merged: https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/520

**Comment by nacho** [comment]
Awesome! thanks @danyeaw

**Pull request merged**